py/server/deephaven/experimental/iceberg.py

```diff
@@ -0,0 +1,251 @@
+        """
+        Reads the table from the catalog using the provided instructions. Optionally, a snapshot id can be provided to
+        read a specific snapshot of the table.
+
+        Args:
+            table_identifier (str): the table to read.
+            instructions (Optional[IcebergInstructions]): the instructions for reading the table. These instructions
+                can include column renames, table definition, and specific data instructions for reading the data files
+                from the provider. If omitted, the table will be read with default instructions.
+            snapshot_id (Optional[int]): the snapshot id to read; if omitted the most recent snapshot will be selected.
+
+        Returns:
+            Table: the table read from the catalog.
+        """
+
+        if instructions is not None:
+            instructions_object = instructions.j_object
+        else:
+            instructions_object = _JIcebergInstructions.DEFAULT
+
+        if snapshot_id is not None:
+            return Table(self.j_object.readTable(table_identifier, snapshot_id, instructions_object))
+        return Table(self.j_object.readTable(table_identifier, instructions_object))
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_catalog_adapter
+
+
+def adapter_s3_rest(
+        catalog_uri: str,
+        warehouse_location: str,
+        name: Optional[str] = None,
+        region_name: Optional[str] = None,
+        access_key_id: Optional[str] = None,
+        secret_access_key: Optional[str] = None,
+        end_point_override: Optional[str] = None
+) -> IcebergCatalogAdapter:
+    """
+    Create a catalog adapter using an S3-compatible provider and a REST catalog.
+
+    Args:
+        catalog_uri (str): the URI of the REST catalog.
+        warehouse_location (str): the location of the warehouse.
+        name (Optional[str]): a descriptive name of the catalog; if omitted the catalog name is inferred from the
+            catalog URI.
+        region_name (Optional[str]): the S3 region name to use; If not provided, the default region will be
+            picked by the AWS SDK from 'aws.region' system property, "AWS_REGION" environment variable, the
+            {user.home}/.aws/credentials or {user.home}/.aws/config files, or from EC2 metadata service, if running in
+            EC2.
+        access_key_id (Optional[str]): the access key for reading files. Both access key and secret access key must be
+            provided to use static credentials, else default credentials will be used.
+        secret_access_key (Optional[str]): the secret access key for reading files. Both access key and secret key
+            must be provided to use static credentials, else default credentials will be used.
+        end_point_override (Optional[str]): the S3 endpoint to connect to. Callers connecting to AWS do not typically
+            need to set this; it is most useful when connecting to non-AWS, S3-compatible APIs.
```

Comment:
What is the default?  What happens with non-AWS and it is not set?

Reply:
If not set, we will not override the s3 endpoints and the file paths from the metadata will not be rewritten.